### PR TITLE
Add try number to StatusBox

### PR DESCRIPTION
**Pull request by curlup**
Tiny UI improvement: 
If there are multiple tries of a task, show the count over the task instance status box in the DAG grid view.

This improves visibility into dag history in the grid view. 

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by bbovenzi** [comment]
Can you include a screenshot please?

File: airflow/www/static/js/dag/StatusBox.tsx [hunk 1]
```diff
@@ -50,8 +52,14 @@ export const StatusWithNotes = ({
...
       borderRadius="2px"
       borderWidth={state ? 0 : 1}
+      color="white"
+      fontSize={boxSizePx}
+      textAlign="center"
+      verticalAlign="middle"
       {...rest}
-    />
+    >
+      {!!tryNumber && tryNumber > 1 && tryNumber}
```
Comment (amoghrajesh) [inline airflow/www/static/js/dag/StatusBox.tsx]: Will `{tryNumber > 1 ? tryNumber : null}` be a simpler check?

**Review by bbovenzi — CHANGES_REQUESTED** [review]
Screenshot 2024-07-30 at 1 37 53 PM

The number doesn't fit well and this won't work for double digit try numbers. If we really want this perhaps we could do an icon instead. I don't know if a redo icon would be appropriate. https://react-icons.github.io/react-icons/search/#q=redo

**Comment by curlup** [comment]
thanks for the feedback

yes, ui is not my strong suit, sorry =\
do you really need to see 2digit retries? is it okay to show only up to 9 maybe? the squares are pretty small, and probably more than 9 retries is rare-ish? i wonder if it's worth it basically. 

> number doesn't fit well

Just to be clear ^ I for sure think fixing this is worth while ofc =)

**Comment by bbovenzi** [comment]
> thanks for the feedback
> 
> yes, ui is not my strong suit, sorry = do you really need to see 2digit retries? is it okay to show only up to 9 maybe? the squares are pretty small, and probably more than 9 retries is rare-ish? i wonder if it's worth it basically.
> 
> > number doesn't fit well
> 
> Just to be clear ^ I for sure think fixing this is worth while ofc =)

You'd be surprised. People use Airflow in all sorts of ways. Se we do have to make sure it can handle >9 tries.

**Comment by curlup** [comment]
Okay, makes sense, any hints on how to make these 2+digit numbers to appear in that run box? I mean the font size to fit in becomes pretty small pretty quickly

**Comment by bbovenzi** [comment]
So I would again recommend for an icon that tells you a task has multiple tries. but to see how many. You need to hover over it to see the tooltip.

**Comment by pierrejeambrun** [comment]
I agree, a visual indicator for multiple tries + a tooltip on hover to display the actual number would be cool. (And would work with any try numbers, 10+, 100+ 1000+)

**Comment by pierrejeambrun** [comment]
Closing as https://github.com/apache/airflow/pull/42152 was merged and implements the same I believe.

**Pull request closed without merging**